sympy/integrals/risch.py

```diff
@@ -680,6 +691,14 @@ def decrement_level(self):
         return None
 
 
+def update_sets(seq, atoms, func):
+    s = set(seq)
+    s = atoms.intersection(s)
```

Comment:
That function could use a docstring -- not necessary for this PR, but I would suggest:

```
def _update_atoms(seq, atoms, func):
    """Return an unordered list of the elements of ``atoms`` which are
    either in seq or pass the filtering by ``func``.

    Examples
    ========

    >>> from sympy.integrals.risch import _update_atoms
    >>> seq = (1, 2, 4)
    >>> atoms = set((4, 5, 6))
    >>> func = lambda x: x % 2
    >>> update_atoms((1, 2, 4), set((4, 5, 6)), func)
    [4, 5]
    >>> all(i in seq or func(i) for i in _)
    True
    """
    new = atoms.intersection(seq)  # keep and don't filter if in seq
    filtered = filter(func, atoms - new)  # filter others
    new.update(filtered)  # combine
    return list(new)
```